### DISCUSSION TITLE
Filament styled components in subheading

### DISCUSSION THREAD

**Discussion by lucacastelnuovo** [filamentphp/filament#6705] [Help] [answered]
In ViewRecord.php the subheading is customizable by overriding the `getSubHeading()` method.

```php
// ...

    protected function getSubheading(): string | Htmlable | null
    {
        return new HtmlString(<<<HTML
            <div class="inline-flex items-center justify-center space-x-1 rtl:space-x-reverse min-h-6 px-2 py-0.5 text-sm font-medium tracking-tight rounded-xl whitespace-nowrap text-success-700 bg-success-500/10">
                <svg class="w-4 h-4" xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" aria-hidden="true">
                    <path stroke-linecap="round" stroke-linejoin="round" d="M14.752 11.168l-3.197-2.132A1 1 0 0010 9.87v4.263a1 1 0 001.555.832l3.197-2.132a1 1 0 000-1.664z"></path>
                    <path stroke-linecap="round" stroke-linejoin="round" d="M21 12a9 9 0 11-18 0 9 9 0 0118 0z"></path>
                </svg>

            <span class="">{$this->record->state->label()}</span>

            </div>
        HTML);
    }
```

Would it be possible to use the badgecolumn here?
Something like this:

```php
BadgeColumn::make('')
  ->label(__('Status'))
  ->icon('heroicon-o-play')
  ->color('success')
  ->placeholder('Actief')
  ->toHtml()
```

When doing this I get the following error: `Filament\Tables\Columns\Column::queriesRelationships(): Argument #1 ($record) must be of type Illuminate\Database\Eloquent\Model, null given, called in vendor/filament/tables/src/Columns/Concerns/HasState.php on line 72 (View: vendor/filament/tables/resources/views/columns/badge-column.blade.php).`

Which is understandable but is there a way to make this work?

**Comment by danharrin** [comment, marked as answer]
I don't really understand why you would want a table column inside a subheading? You have complete control over the HTML, but you can't just put something, that was built for the table, outside of a table

**Reply by lucacastelnuovo** [reply]
Yeah, you are right. It was maybe a bit too hacky.
